tests/core/app.test.ts

```diff
@@ -683,6 +685,23 @@ describe('Subapps', () => {
 
     await fetch('/').expect(200, 'Hello World!')
   })
+  it('sub-app mounted on root and subpath', async () => {
```

Comment:
Can you rename this test to reflect the actual test?

Reply:
I don't like that name either - just trying my best to follow the convention that the similar tests follow.

If you give me the particular name you want me to use, sure. 

I don't fancy playing that guessing game presently 😅

Reply:
Maybe something along the lines of `should search for routes in other subapps even if a subapp mounted on '/' is present`. What do you reckon?

Reply:
Aye, I catch your drift. I might go for something like this:
```ts
describe("when a subapp is mounted on '/'", () => {
  it("should continue middleware execution when subapp middleware is exhausted", () => { ... })
  // other related tests
})
```

Reply:
Great! Are there other test cases which you can group under `when a subapp is mounted on '/'`? It wouldn't make sense to have a test suite with only 1 test.

Reply:
> Great! Are there other test cases which you can group under `when a subapp is mounted on '/'`? It wouldn't make sense to have a test suite with only 1 test.

There are, so this is moot, but regardless - I disagree, using `describe` to provide context to a single testcase is a common practise. See Mocha's `context` ([example](https://mochajs.org/#bdd>))

Reply:
Cool.